### ISSUE TITLE
mozjpeg as feature

### ISSUE THREAD

**Issue by godofdream** [image-rs/image#1350]
jpeg encoding is currently a relatively simple implementation. 
For better encoding mozjpeg is a valid solution.
It would be nice to add mozjpeg as feature for compression of jpegs.
There is mozjpeg-sys as ffi-binding to the official library, and a high level binding called mozjpeg, however both don't support this crate.

**Comment by kornelski** [comment]
Alternatively, I'd suggest looking into porting jpegli to Rust.

**Comment by fintelia** [comment]
Yeah, for this crate we probably don't want to add the complexity of another C based dependency (it is hard enough dealing with the ones we already have!) but a Rust port would be far more appealing

**Comment by 197g** [comment]
Maybe in [`image-extra`](https://github.com/image-rs/image-extras)? At least the bindings, how to enable them and which hooks the crate should add is f course another question better tracked there.
